Packs/Code42/Integrations/Code42/Code42.py

```diff
@@ -15,509 +17,1178 @@
+        file_results = CommandResults(
+            outputs_prefix="File", outputs_key_field=None, outputs=file_context
+        )
+        return code42_results, file_results
+
     else:
-        return 'No results found', {}, {}
+        return CommandResults(
+            readable_output="No results found",
+            outputs={"Results": []},
+            outputs_key_field="EventID",
+            outputs_prefix="Code42.SecurityData",
+            raw_response={},
+        )
+
+
+def user_create_command(client, args):
+    org_name = args.get("orgname")
+    username = args.get("username")
+    email = args.get("email")
+    res = client.create_user(org_name, username, email)
+    outputs = {
+        "Username": res.get("username"),
+        "UserID": res.get("userUid"),
+        "Email": res.get("email"),
+    }
+    readable_outputs = tableToMarkdown("Code42 User Created", outputs)
+    return CommandResults(
+        outputs_prefix="Code42.User",
+        outputs_key_field="UserID",
+        outputs=outputs,
+        readable_output=readable_outputs,
+        raw_response=res,
+    )
+
+
+def user_block_command(client, args):
+    username = args.get("username")
+    user_id = client.block_user(username)
+    outputs = {"UserID": user_id}
+    readable_outputs = tableToMarkdown("Code42 User Blocked", outputs)
+    return CommandResults(
+        outputs_prefix="Code42.User",
+        outputs_key_field="UserID",
+        outputs=outputs,
+        readable_output=readable_outputs,
+        raw_response=user_id,
+    )
+
+
+def user_unblock_command(client, args):
+    username = args.get("username")
+    user_id = client.unblock_user(username)
+    outputs = {"UserID": user_id}
+    readable_outputs = tableToMarkdown("Code42 User Unblocked", outputs)
+    return CommandResults(
+        outputs_prefix="Code42.User",
+        outputs_key_field="UserID",
+        outputs=outputs,
+        readable_output=readable_outputs,
+        raw_response=user_id,
+    )
+
+
+def user_deactivate_command(client, args):
+    username = args.get("username")
+    user_id = client.deactivate_user(username)
+    outputs = {"UserID": user_id}
+    readable_outputs = tableToMarkdown("Code42 User Deactivated", outputs)
+    return CommandResults(
+        outputs_prefix="Code42.User",
+        outputs_key_field="UserID",
+        outputs=outputs,
+        readable_output=readable_outputs,
+        raw_response=user_id,
+    )
+
+
+def user_reactivate_command(client, args):
+    username = args.get("username")
+    user_id = client.reactivate_user(username)
+    outputs = {"UserID": user_id}
+    readable_outputs = tableToMarkdown("Code42 User Reactivated", outputs)
+    return CommandResults(
+        outputs_prefix="Code42.User",
+        outputs_key_field="UserID",
+        outputs=outputs,
+        readable_output=readable_outputs,
+        raw_response=user_id,
+    )
+
+
+def download_file_command(client, args):
```

Comment:
Realizing we might want a `filename` param here so that we can save the file with its correct name from Playbook. It could be options and we could use the hash string as the fallback.

Reply:
Also just learned that the actual filename is part of the response headers, so if they don't provide a filename, that could be the default